rules/utils/is-function-self-used-inside.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+const {findVariable} = require('eslint-utils');
+
+const getReferences = (scope, nodeOrName) => {
+	const {references = []} = findVariable(scope, nodeOrName) || {};
+	return references;
+};
+
+/**
+Check if `this`, `arguments`, or `id` of function is used inside it self.
```

Comment:
```suggestion
Check if `this`, `arguments`, or `id` of the function is used inside of itself.
```

Reply:
Not clear from reading here what the "id" of the function is.

Reply:
Changed to ``Check if `this`, `arguments`, or the function name is used inside of itself.``